### PR TITLE
AutoML: add "search order" to rankings and full_rankings

### LINKED ISSUE #1841: AutoML: add "search order" to rankings and full_rankings
Our `AutoMLSearch` object has a `search_order` field which indicates the pipeline IDs of the pipelines automl has tried, in the order in which automl tried them.

We should add that as another column to the `rankings` and `full_rankings` tables. That would make it easy for users to match the pipelines up with the progress on the search iteration plot. It could also be a coarse way to show whether or not tuning added value--if the best pipeline(s) have low `search_order` numbers, that means the earlier rounds outperformed the later ones.

### PR DESCRIPTION
**Pull request by angela97lin**
Closes #1841

### CODE REVIEW

File: evalml/tests/automl_tests/test_automl.py [hunk 1]
```diff
@@ -160,6 +160,7 @@ def test_search_results(X_y_regression, X_y_binary, X_y_multi, automl_type, obje
                 np.dtype("float64"),
                 np.dtype("bool"),
                 np.dtype("O"),
```
Comment (jeremyliweishih) [inline evalml/tests/automl_tests/test_automl.py:162]: do any of these tests check if "search_order" has the correct values?

**Review by jeremyliweishih — APPROVED** [review]
Looks good but just have one testing question. If that case isn't covered we should add it.

File: evalml/tests/automl_tests/test_automl.py (hunk 1 above)
Comment (angela97lin) [reply evalml/tests/automl_tests/test_automl.py:162]: @jeremyliweishih Line 109 checks: ``assert automl.results["search_order"] == [0, 1]`` :)

File: evalml/automl/automl_search.py [hunk 2]
```diff
@@ -1262,11 +1261,14 @@ def full_rankings(self):
...
         ]
+        full_rankings_cols = pipeline_results_cols + ["search_order"]
+
         if not self._results["pipeline_results"]:
             return pd.DataFrame(columns=full_rankings_cols)
 
         rankings_df = pd.DataFrame(self._results["pipeline_results"].values())
-        rankings_df = rankings_df[full_rankings_cols]
+        rankings_df = rankings_df[pipeline_results_cols]
+        rankings_df["search_order"] = pd.Series(self._results["search_order"])
```
Comment (freddyaboulton) [inline evalml/automl/automl_search.py:1271]: Does this mean search order is the last column? Might make more sense to put it next to the pipeline name?

**Review by freddyaboulton — APPROVED** [review]
@angela97lin Looks good!

**Review by bchen1116 — APPROVED** [review]
LGTM!

File: evalml/automl/automl_search.py (hunk 2 above)
Comment (angela97lin) [reply evalml/automl/automl_search.py:1271]: Yeah :'( we get the other info from pipeline_results so I didn't want to break that up, but I think this makes more sense too

**Pull request merged**